### ISSUE TITLE
Support hivemind's condenser_api

### ISSUE THREAD

**Issue by relativityboy** [steemit/steem-js#305]

**Comment by jnordberg** [comment]
I'm not a huge fan of this, it makes an already convoluted api more so. I'd prefer if we move to calling all apis with their method names directly, e.g. `api.call('get_dynamic_global_properties')` and `api.call('hive_api.condenser_api.get_dynamic_global_properties')` if you want to speak with hive.

If we want to retire an steemd method and let hive take over we can do that routing in jussi.

**Comment by bnchdrff** [comment]
is there any downside to having jussi take care of this? it seems like the ideal place to do it, even in the case of running jussi for local dev & setting up weird temporary routing.

**Comment by relativityboy** [comment]
The issues are that 
1. **jussi shouldn't care about this stuff** We want jussi 1.0 to remain dumb with regard to specific api needs (no api route should need "special" handling ex: request restructuring)
2. **Hive needs to support a 'pure legacy' endpoint**
  1. We may need to switch between hive's `/legacy` (non-appbase style calls), hive `/` and steemd depending on traffic and performance. All these have different mixes of rpc structure, method prefix, and method name. (this 3rd property is the sometime replacement of the method name with 'call')
3. This change keeps us from breaking 3rd party code until such time as we can move to our new api client.

I'm definitely in favor of something like `api.call('hive.get_dynamic_global_properties')` once we eliminate the need to mask switching between baked-in rpc calls like 

```
{
   method: 'someservice.call',
   params: ['blah', 'get_dynamic_global_properties', {realparamsgohere}]
}
```
vs
```
{
   method: 'someservice.get_dynamic_global_properties',
   params: {realparamsgohere}
}
```
on a method by method basis.

**Comment by jnordberg** [comment]
We should never need to do request restructuring, here or in jussi. When is that needed?

As for not breaking backwards compatibility when we move some steemd calls to hive that will happen in jussi

**Comment by relativityboy** [comment]
> We should never need to do request restructuring, here or in jussi. When is that needed?
I agree with this in principle. If we have to switch between the "old format" and the new quickly (imagining some performance issue, outage, or bottleneck) we have a way of going back.

> As for not breaking backwards compatibility when we move some steemd calls to hive that will happen in jussi

You're talking about the routing? My understanding is that `condenser_api.` and `` are supposed to point to steemd, even after hive goes live.

At some point in the nearish future we have a PR that let's us eliminate this type of call entirely.
```
{
   method: 'someservice.call',
   params: ['blah', 'get_dynamic_global_properties', {realparamsgohere}]
}
```

**Comment by jnordberg** [comment]
I'm with you on that @roadscape, my point is that we don't need the client-side method routing this introduces to support hive. All we need to do is use `steem.api.call`. E.g. switching between `steem.api.call('condenser_api.get_followers', 'bar')` and `steem.api.call('hive.condenser_api.get_followers', 'bar')`. Or am I missing something?

**Comment by relativityboy** [comment]
Things have gotten a lot simpler since this issue (and the related PR) were filed. 
Hive's MVP is to have one endpoint only, and Jussi is handling api structure translations.

I'l be revisiting the PR and this issue to see how much I can simplify it, possibly make it go away entirely.

-

@jnordberg philosophically you're spot on. That's definitely the simpler way for us to do it, and where we want to quickly get. 

It also introduces more potential pain for users of the library.
 
Offering this configuration option gives us the ability to change all calls to `api.getFollowers()` of those hydrated functions from `condenser_api.get_followers` to `hive.condenser_api.get_followers` .. or any other hydrated function for that matter - from a single authoritative location. It also allows any 3rd party developers running our library to upgrade their library without having to do modifications to multiple locations in their code, rewrite tests, etc.

**Comment by ety001** [comment]
This repository has been superseded by a complete 2025 rewrite. The new trunk (`master`, starting at v1.1.0) is a ground-up rewrite of `@steemit/steem-js` and the legacy 0.7.x / 0.8.x code line is archived.

As part of that transition, this issue is being closed because:
- It was filed against the legacy codebase, and the code it references no longer exists in the rewritten trunk.
- The new codebase reimplements the affected areas (RPC routing, transaction serialization, auth/signing, crypto, browser builds) from scratch, so this issue cannot be carried over as-is.

If the problem or request still applies to the rewritten codebase (v1.1.0+ on `master`), please open a new issue against the current trunk with details that reproduce on the new version. Thank you for taking the time to report this originally.
